src/CssInliner.php

```diff
@@ -451,8 +454,14 @@ private function normalizeStyleAttributes(\DOMElement $node): void
         $callback = static function (array $propertyNameMatches) use ($declarationBlockParser): string {
             return $declarationBlockParser->normalizePropertyName($propertyNameMatches[0]);
         };
-        $normalizedOriginalStyle = (new Preg())->throwExceptions($this->debug)
-            ->replaceCallback($pattern, $callback, $node->getAttribute('style'));
+        // The safe version is only available in "thecodingmachine/safe" for PHP >= 8.1.
+        if (\function_exists('Safe\\preg_replace_callback')) {
+            $normalizedOriginalStyle = preg_replace_callback($pattern, $callback, $node->getAttribute('style'));
+            \assert(\is_string($normalizedOriginalStyle));
```

Comment:
The `assert` is surely redundant when using the `Safe` version.

Reply:
No, it's not.

The `Safe` functions make sure that errors throw exceptions instead of returning `false`. However, those functions do not provide PHPStan-compatible annotations like "if a string is provided, a string will be returned, and if an array is provided, an array will be returned".

With our own wrapper, we could do without the `assert` as our wrapper does not support arrays, but only strings.

Reply:
So this assert is about the value being a `string` instead of an `array`, not about it not being `false`.

Reply:
Ah, I see.  I've seen it possible to template the type annotations, or put conditionals into them, so maybe a feature request for 'Safe'...

Reply:
> Yes, absolutely. Would you be willing to create the issue (or provide a PR)?

I am not sure of the syntax required, so may need some help.  I have seen such syntax (templates and conditionals), and have used the template syntax once, so am aware it's possible, but not 100% sure how, or what's best to propose...

Reply:
https://github.com/thecodingmachine/safe/issues/703